tests/test_runlib.py

```diff
@@ -726,6 +814,40 @@ def test_normalize_line_endings(self):
       for path in paths:
         os.remove(path)
 
+  def test_wrong_metadata_directory(self):
+    """Test record stop with wrong metadata directory"""
+    if sys.version_info < (3, 0):
+      with self.assertRaises(IOError):
+        in_toto_record_start(self.step_name, [], self.key)
+        in_toto_record_stop(self.step_name, [], self.key,
+            metadata_directory='this-is-not-a-dir')
+    else:
+      with self.assertRaises(FileNotFoundError):
+        in_toto_record_start(self.step_name, [], self.key)
+        in_toto_record_stop(self.step_name, [], self.key,
+            metadata_directory='this-is-not-a-dir')
+      with self.assertRaises(NotADirectoryError):
+        path = tempfile.mkstemp(dir=os.getcwd())[1]
+        in_toto_record_start(self.step_name, [], self.key)
+        in_toto_record_stop(self.step_name, [], self.key,
+            metadata_directory=path)
```

Comment:
Since these lines are only run on py3, you could put them in separate test function and add a decorator to skip on py2.